contracts/modules/v2/StakingPool.sol

```diff
@@ -348,21 +356,55 @@ contract StakingPool is IStakingPool, ERC721 {
     rewardsSharesSupply = _rewardsSharesSupply + newRewardsShares;
   }
 
-  function calculateRewardSharesAmount(
-    uint stakeSharesAmount,
-    uint trancheId
-  ) internal view returns (uint) {
-
-    uint lockDuration = (trancheId + 1) * TRANCHE_DURATION - block.timestamp;
-    uint maxLockDuration = TRANCHE_DURATION * 8;
-
-    // TODO: determine extra rewards formula
-    return
-      stakeSharesAmount
-      * REWARDS_SHARES_RATIO
-      * lockDuration
-      / REWARDS_SHARES_DENOMINATOR
-      / maxLockDuration;
+  function getTimeLeftOfTranche(uint trancheId, uint blockTimestamp) internal pure returns (uint) {
+    uint endDate = (trancheId + 1) * TRANCHE_DURATION;
+    if (endDate > blockTimestamp) {
+      return endDate - blockTimestamp;
+    }
+    return 0;
+  }
+
+  /// Calculates the amount of new reward shares based on the initial and new stake shares and
+  /// tranche.
+  ///
+  /// @param initialStakeShares  The amount of stake shares that the deposit is already entitled
+  ///                            to.
+  /// @param newStakeShares      The amount o new stake shares that the deposit will entitled to
+  ///                            on top of the existing ones.
+  /// @param initialTrancheId    The initial id of the tranche that defines the deposit period.
+  /// @param newTrancheId        The new id of the tranche that will define the deposit period.
+  /// @param blockTimestamp      The timestamp of the block when the new shares are recalculated.
+  function calculateNewRewardShares(
+    uint initialStakeShares,
+    uint newStakeShares,
```

Comment:
So `newStakeShares` is on top of `initialStakeShares`

`extraStakeShares` could be more clear

Reply:
The term "newShares" is already used throughout the contract and I do agree it doesn't clearly tell whether it is the final state after the transaction or something that only adds up to the initial state resulting in the final state.

I'd suggest `newStakeShares` -> `stakeSharesIncrease` and apply this pattern it consistently.

Reply:
`stakeSharesIncrease` good enough